### PR TITLE
Add release conductor notes and workflow

### PR DESCRIPTION
**Pull request by rezrah**
## Summary

Adds release conductor notes, checklist and automated reminder about the checklist to Release Tracking PRs.

Adapted from a [similar checklist added to PRC](https://github.com/primer/react/pull/2015).

## What should reviewers focus on?

- Please review docs and ask any questions if you have any

## Steps to test:

1. Scroll down to see the rendered version of the automated reminder

Screenshot 2024-07-19 at 17 58 13

### CODE REVIEW

**Review by raytalks — APPROVED** [review]
✨

File: .github/release_template.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    {insert screenshot}
+
+  Release testing status: Ready to release 🟢 | Paused 🟡 🔴 (choose one and provide reason)
+  ```
+
+### 🚢 After merge
+
+- [ ] Issue release comms in `primer brand` Slack channel
+
+- [ ] Add the preview deployment's link to releases as a way of permalinking to old version's docs. [Example](https://github.com/primer/brand/releases/tag/%40primer%2Freact-brand%400.34.3)
```
Comment (danielguillan) [inline .github/release_template.md:78]:
```suggestion
- [ ] Issue release comms in `primer brand` Slack channel
- [ ] Add the preview deployment's link to releases as a way of permalinking to old version's docs. [Example](https://github.com/primer/brand/releases/tag/%40primer%2Freact-brand%400.34.3)
```

**Review by danielguillan — APPROVED** [review]

File: .github/workflows/release_canary.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ on:
 
 jobs:
   release-canary:
-    if: ${{ github.repository == 'primer/brand' }}
+    if: false
```
Comment (joshfarrant) [inline .github/workflows/release_canary.yml:12]: Was this just changed for testing purposes?

File: .github/workflows/release_candidate_checklist.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          node-version: 20
+
+      - name: Get or Create Comment
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const fs = require('fs')
+            const body = await fs.readFileSync('.github/release_template.md', 'utf8')
+            console.log('body', body)
```
Comment (joshfarrant) [inline .github/workflows/release_candidate_checklist.yml:28]: Is this intentionally left in, or just debugging?

**Review by joshfarrant — APPROVED** [review]
Nice! A couple of small questions, but really happy this is being added 💖

File: .github/workflows/release_candidate_checklist.yml (hunk 3 above)
Comment (rezrah) [reply .github/workflows/release_candidate_checklist.yml:28]: Yeah, this is just debugging and will be removed before merging 👍

File: .github/workflows/release_canary.yml (hunk 2 above)
Comment (rezrah) [reply .github/workflows/release_canary.yml:12]: Yeah, it's just to stop publishing canaries which are killing my inbox 😄

**Pull request merged**